StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Elements/TextField/TextFieldElement+Card.swift

```diff
@@ -288,7 +295,12 @@ extension TextFieldElement {
                 }
                 // Is the date expired?
                 guard let expiryDate = CardExpiryDate(text), !expiryDate.expired() else {
-                    return .invalid(Error.invalid)
+                    if !isEditable {
+                        return .invalid(Error.expired)
+                    }
+                    else {
+                        return .invalid(Error.invalid)
+                    }
```

Comment:
Using `isEditable` probably isn't the best way to determine which error we should show, they should be independent.

You can break this up into

````swift
guard let expiryDate = CardExpiryDate(text) else {
   return .invalid(Error.invalid)
}

guard !expiryDate.expired() else {
   return .invalid(Error.expired)
}